components/Header.tsx

```diff
@@ -1,11 +1,105 @@
 import Link from "next/link";
+import Image from "next/image";
+import { useRouter } from 'next/router';
+import { useEffect, useState } from "react";
 
 const Header = () => {
+    const router = useRouter();
+    const currentPath = router.pathname;
+
+    const isActive = (path: string): string => {
```

Comment:
You don't need the second `string` type here because the return can only be `"border-project-green bg-project-green font-bold" | "group"`
TypeScript will automatically detect a type in situations like this. You can check this by removing the second `string` type and hovering over `isActive`